### PR TITLE
support both yaml and yml postfix

### PR DESCRIPTION
**Pull request by Stephen1993**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/_utils/flow_utils.py [hunk 1]
```diff
@@ -87,13 +88,19 @@ def resolve_flow_path(
...
+            flow_file_name = flow_name.replace(".yaml", suffix)
+            if (flow_folder / flow_file_name).is_file():
+                flow_file_num += 1
+                flow_file = flow_file_name
+
+        if flow_file_num > 1:
             raise ValidationException(
-                f"Both {FLOW_DAG_YAML} and {FLOW_FLEX_YAML} exist in {flow_path}. "
-                f"Please specify a file or remove the extra YAML.",
+                f"Both {FLOW_DAG_YAML}/{FLOW_DAG_YAML.replace('.yaml', '.yml')} "
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/_utils/flow_utils.py:101]: how about we print the file we found in the directory?

File: src/promptflow/tests/sdk_cli_test/e2etests/test_flow_test.py [hunk 2]
```diff
@@ -275,7 +275,13 @@ def test_eager_flow_test(self):
 
     def test_eager_flow_test_with_yaml(self):
         clear_module_cache("entry")
-        flow_path = Path(f"{EAGER_FLOWS_DIR}/simple_with_yaml/").absolute()
+        flow_path = Path(f"{EAGER_FLOWS_DIR}/simple_with_yml/").absolute()
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_flow_test.py:278]: Why is it still `simple_with_yml` while the test title says `with_yaml`?

Comment (Stephen1993) [reply src/promptflow/tests/sdk_cli_test/e2etests/test_flow_test.py:278]: Good catch, I accidentally modified this.

**Review by 0mza987 — APPROVED** [review]

File: src/promptflow-core/promptflow/_utils/flow_utils.py [hunk 3]
```diff
@@ -89,18 +89,18 @@ def resolve_flow_path(
...
-        if flow_file_num > 1:
+        if len(flow_file_list) == 1:
+            flow_file = flow_file_list[0]
+        elif len(flow_file_list) > 1:
             raise ValidationException(
-                f"Both {FLOW_DAG_YAML}/{FLOW_DAG_YAML.replace('.yaml', '.yml')} "
-                f"and {FLOW_FLEX_YAML}/{FLOW_FLEX_YAML.replace('.yaml', '.yml')} exist in {flow_path}. "
-                f"Please specify a file or remove the extra YAML/YML.",
+                f"Multiple files {', '.join(flow_file_list)} exist in {flow_path}. "
+                f"Please specify a file or remove the extra YAML and YML.",
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/_utils/flow_utils.py:103]: YAML and YML -> YAML or YML?

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-core/promptflow/_utils/flow_utils.py (hunk 3 above)
Comment (Stephen1993) [reply src/promptflow-core/promptflow/_utils/flow_utils.py:103]: There is already an or in front, is it a bit inconvenient to use or here?
image

**Review by wangchao1230 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**